STAC-22594  Documented how to verify signature of docker images (#404)
rancher/stackstate-product-docs: docs/latest/modules/en/pages/setup/security/verify-image-signatures.adoc

[,bash]
----
cosign verify \
  --certificate-oidc-issuer=https://token.actions.githubusercontent.com \
  --certificate-identity-regexp='^https://github.com/StackVista/.*' \
  registry.rancher.com/suse-observability/<image>:<tag>
----

Replace `<image>:<tag>` with the image and tag you want to verify, for example `clickhouse:26.5.1-so8`.

A successful verification prints a `Verification for ...` banner followed by the signed claims, and the command exits with status `0`. If the image is unsigned, or was signed by anything other than our CI, the command fails with a non-zero exit code and an error.

=== Scoping to a specific repository

The default command accepts a signature from *any* workflow in the `StackVista` GitHub organization. For a tighter check, scope the pattern to the specific source repository behind an image, for example `stackstate`:

[,bash]
----
cosign verify \
  --certificate-oidc-issuer=https://token.actions.githubusercontent.com \
  --certificate-identity-regexp='^https://github.com/StackVista/stackstate/.*' \
  registry.rancher.com/suse-observability/<image>:<tag>
----

> 

Author: deleted

====

If you do need to inspect the exact identity that signed a specific image — for example while troubleshooting — it is embedded in the signing certificate, not in `cosign verify`'s own output, so extract it directly:

[,bash]
----
cosign download signature registry.rancher.com/suse-observability/<image>:<tag> \
  | jq -r '.verificationMaterial.certificate.rawBytes' \
  | base64 -d \
  | openssl x509 -inform der -noout -text \
  | grep -A1 'Subject Alternative Name' \
  | tail -1 \
  | sed 's/.*URI://'
----

`cosign download signature` only reads what is there — it does not check trust, so treat its output as informational, and always confirm the image with an actual `cosign verify` call.

> == Mirroring images to a private or air-gapped registry

Author: deleted

= Verify container image signatures
:page-languages: [en, de, es, fr, ja, pt, zh]
:revdate: 2026-08-26
:page-revdate: {revdate}
:description: How to verify the cryptographic signature of {stackstate-product-name} container images

== Overview

> {stackstate-product-name} container images are cryptographically signed during the build pipeline using https://www.sigstore.dev/[Sigstore] keyless signing (`cosign`). Verifying a signature confirms that an image was genuinely built by our CI from a specific source repository, workflow, and branch or tag — before you pull it into a cluster.

Reviewer: ```suggestion
{stackstate-product-name} container images are cryptographically signed during the build pipeline using https://www.sigstore.dev/[Sigstore] keyless signing (`cosign`). Verifying a signature confirms that an image was genuinely built by SUSE from a specific source repository, workflow, and branch or tag — before you pull it into a cluster.
```

> {stackstate-product-name} container images are cryptographically signed during the build pipeline using https://www.sigstore.dev/[Sigstore] keyless signing (`cosign`). Verifying a signature confirms that an image was genuinely built by our CI from a specific source repository, workflow, and branch or tag — before you pull it into a cluster.

Reviewer 2: ```suggestion
{stackstate-product-name} container images are cryptographically signed during the build pipeline using link:https://www.sigstore.dev/[Sigstore] keyless signing (`cosign`). Verifying a signature confirms that an image was genuinely built by our CI from a specific source repository, workflow, and branch or tag — before you pull it into a cluster.
```

Signing is keyless: there is no public key file to download or manage. Instead, each signature is backed by a short-lived certificate tied to the GitHub Actions job that produced it, and a public, tamper-evident record of the signing event.

> Image signing was added to our build pipeline recently. Images published before this was rolled out do not carry a signature — verifying one of those tags fails with `no signatures found`, which does not indicate a problem with the image itself. If you rely on signature verification, pull a current tag.

Reviewer: ```suggestion
Image signing was introduced for platform images in release vX.Y.Z and for agent images in release vA.B.C.. Images published before this was rolled out do not carry a signature — verifying one of those tags fails with `no signatures found`, which does not indicate a problem with the image itself. If you rely on signature verification, pull a current tag.
```

== Prerequisites

Install `cosign` (the Sigstore client):

[,bash]
----
# macOS
brew install cosign

# Linux, or any platform: download a release binary
# https://github.com/sigstore/cosign/releases
----

== Verify a signature

Run `cosign verify` against the image reference, pinning the OIDC issuer to GitHub Actions and matching the certificate identity to the `StackVista` GitHub organization:

[,bash]
----
cosign verify \
  --certificate-oidc-issuer=https://token.actions.githubusercontent.com \
  --certificate-identity-regexp='^https://github.com/StackVista/.*' \
  registry.rancher.com/suse-observability/<image>:<tag>
----

Replace `<image>:<tag>` with the image and tag you want to verify, for example `clickhouse:26.5.1-so8`.

A successful verification prints a `Verification for ...` banner followed by the signed claims, and the command exits with status `0`. If the image was not signed by SUSE, the command fails with a non-zero exit code and an error.

== Troubleshooting

`no signatures found`::

> This is most commonly one of: an older tag published before we started signing images (see <<_overview,Overview>>); or a mistyped registry, repository, or tag.

Reviewer 2: ```suggestion
This is either an older tag published before we started signing images (see <<_overview,Overview>>) or a mistyped registry, repository, or tag.
```
